### ISSUE TITLE
Research city state gem upgrade for use in setting location based on buckets

### ISSUE THREAD

**Issue by dboyer** [Iridescent-CM/technovation-app#4563]
Goals:

1. Determine if we want to use the city-state gem
2. Determine how to get rid of our forked version and instead use the version through the gem source

**Comment by viviancan** [comment]
btw here is the[ our fork of the gem](https://github.com/Iridescent-CM/city-state)

**Comment by viviancan** [comment]
I think the city state gem will work. As a proof of concept, I removed the forked version and added the original city-state gem. I created an maxmind account using the engineering email. Credentials & license key in password manager.

Following [this blog post](https://learnwithdaniel.com/2015/02/citystate-list-of-countries-cities-and-states-ruby/) I created a basic rails form in place of the original location vue form to test out the idea. I think this path forward will work as expected. I did not try to "update" the city-state data yet, but the docs state the data can be updated using the following

```
CS.set_license_key('MY_KEY')

CS.update
```

Based on this information, I am moving this ticket into the review/QA and we can determine next steps! 

[Private Zenhub Video](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBNFBQQXc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--d83ed4269e5cb823e5ea0f774a46794e142ac821/CleanShot%202024-03-11%20at%2018.04.40.mp4)

**Comment by shaun-technovation** [comment]
I started looking at the city-state gem too, looks like it's a good out-of-the-box option.

If we want to add additional location data (which the gem supports)... I was able to add an additional state, but I wasn't able to add an additional city, looks like other people are running into this issue to:
- https://github.com/thecodecrate/city-state/issues/67
- https://github.com/thecodecrate/city-state/issues/43

That's one downside I'm seeing is that the gem maintainer(s) aren't very active/responsive. But, perhaps this gem will work if we don't need any customizations. 🤔
